src/mongocxx/options/find_one_and_replace.hpp

```diff
@@ -19,9 +19,11 @@
 #include <cstdint>
 
 #include <bsoncxx/document/view.hpp>
-#include <mongocxx/options/find_one_and_modify.hpp>
 #include <bsoncxx/stdx/optional.hpp>
```

Comment:
Yeah. Generally a principle I support but maybe working against us here. If we are really only including the bsoncxx/stdx/xyz.hpp headers we need and then individually pulling in the types with using, I think it is something we could live with.